image/directory/directory_dest.go

```diff
@@ -20,20 +20,73 @@ import (
 	"go.podman.io/storage/pkg/fileutils"
 )
 
-const version = "Directory Transport Version: 1.1\n"
+const (
+	versionPrefix = "Directory Transport Version: "
+)
+
+// version represents a parsed directory transport version
+type version struct {
+	major int
+	minor int
+}
+
+// Supported versions
+// Write version file based on digest algorithm used.
+// 1.1 for sha256-only images, 1.2 otherwise.
+var (
+	version1_1          = version{major: 1, minor: 1}
+	version1_2          = version{major: 1, minor: 2}
+	maxSupportedVersion = version1_2
+)
+
+// String formats a version as a string suitable for writing to the version file
+func (v version) String() string {
+	return fmt.Sprintf("%s%d.%d\n", versionPrefix, v.major, v.minor)
+}
+
+// parseVersion parses a version string into major and minor components.
+// Returns an error if the format is invalid.
+func parseVersion(versionStr string) (version, error) {
+	var v version
+	expectedFormat := versionPrefix + "%d.%d\n"
+	n, err := fmt.Sscanf(versionStr, expectedFormat, &v.major, &v.minor)
+	if err != nil || n != 2 || versionStr != fmt.Sprintf(expectedFormat, v.major, v.minor) {
+		return version{}, fmt.Errorf("invalid version format")
+	}
+	return v, nil
+}
+
+// isGreaterThan returns true if v is greater than other
+func (v version) isGreaterThan(other version) bool {
```

Comment:
(Absolutely non-blocking: returning values with the same semantics as `cmp.Compare` could make other interoperability with `cmp` easier; e.g. this could be `cmp.Or(cmp.Compare( … majors), cmp.Compare( … minors)`. That would not be any better for the two callers that we do have, though.